### PR TITLE
feat(core): support nx-cloud as non-default task runner

### PR DESCRIPTION
**Pull request by StalkAltan**

### CODE REVIEW

File: packages/nx/src/utils/nx-cloud-utils.ts [hunk 1]
```diff
@@ -1,13 +1,27 @@
 import { NxJsonConfiguration, readNxJson } from '../config/nx-json';
 
-export function isNxCloudUsed(nxJson: NxJsonConfiguration) {
-  return (
+// Returns tuple [isCloudUsed, cloudConfigurationName]
+export function isNxCloudUsed(nxJson: NxJsonConfiguration): [boolean, string] {
```
Comment (FrozenPandaz) [inline packages/nx/src/utils/nx-cloud-utils.ts:4]: I think this function should only return `boolean`.

1. It's weird otherwise
2. There's no real magic going on here which can't be moved into `getCloudOptions` or somewhere else.

File: packages/nx/src/tasks-runner/run-command.ts [hunk 2]
```diff
@@ -499,21 +493,10 @@ export function getRunnerOptions(
     ...nxArgs,
   };
 
-  // NOTE: we don't pull from env here because the cloud package
-  // supports it within nx-cloud's implementation. We could
-  // normalize it here, and that may make more sense, but
-  // leaving it as is for now.
-  if (nxJson.nxCloudAccessToken && isCloudDefault) {
```
Comment (FrozenPandaz) [inline packages/nx/src/tasks-runner/run-command.ts:506]: This seems unrelated? should this actually have changed?

With these changes, this would...  add this info to all runners, even non-nx-cloud runners, if it's in `nx.json`

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/nx/src/command-line/migrate/migrate.ts [hunk 3]
```diff
@@ -1216,11 +1216,12 @@ async function generateMigrationsJsonAndUpdatePackageJson(
       readNxVersion(originalPackageJson);
 
     try {
+      const [isCloudUsed, _] = isNxCloudUsed(originalNxJson);
```
Comment (meeroslav) [inline packages/nx/src/command-line/migrate/migrate.ts:1219]: The second parameter is not used so there is no need to define it.

```suggestion
      const [isCloudUsed] = isNxCloudUsed(originalNxJson);
```

File: packages/nx/src/command-line/connect/connect-to-nx-cloud.ts [hunk 4]
```diff
@@ -48,7 +48,8 @@ export async function connectToNxCloudIfExplicitlyAsked(
 
 export async function connectToNxCloudCommand(): Promise<boolean> {
   const nxJson = readNxJson();
-  if (isNxCloudUsed(nxJson)) {
+  const [isCloudUsed, _] = isNxCloudUsed(nxJson);
```
Comment (meeroslav) [inline packages/nx/src/command-line/connect/connect-to-nx-cloud.ts:51]: The second parameter is not used so there is no need to define it.

```suggestion
      const [isCloudUsed] = isNxCloudUsed(originalNxJson);
```

File: packages/nx/src/tasks-runner/life-cycles/view-logs-utils.ts [hunk 5]
```diff
@@ -5,7 +5,8 @@ import { output } from '../../utils/output';
 const VIEW_LOGS_MESSAGE = `Hint: Try "nx view-logs" to get structured, searchable errors logs in your browser.`;
 
 export function viewLogsFooterRows(failedTasks: number) {
-  if (failedTasks >= 2 && !isNxCloudUsed(readNxJson())) {
+  const [isCloudUsed, _] = isNxCloudUsed(readNxJson());
```
Comment (meeroslav) [inline packages/nx/src/tasks-runner/life-cycles/view-logs-utils.ts:8]: The second parameter is not used so there is no need to define it.

```suggestion
  const [isCloudUsed] = isNxCloudUsed(readNxJson());
```

File: packages/nx/src/command-line/connect/view-logs.ts [hunk 6]
```diff
@@ -7,8 +7,8 @@ import { readNxJson } from '../../config/nx-json';
 import { connectExistingRepoToNxCloudPrompt } from './connect-to-nx-cloud';
 
 export async function viewLogs(): Promise<number> {
-  const cloudUsed = isNxCloudUsed(readNxJson());
-  if (cloudUsed) {
+  const [isCloudUsed, _] = isNxCloudUsed(readNxJson());
```
Comment (meeroslav) [inline packages/nx/src/command-line/connect/view-logs.ts:10]: The second parameter is not used so there is no need to define it.

```suggestion
  const [isCloudUsed] = isNxCloudUsed(readNxJson());
```

**Review by meeroslav — COMMENTED** [review]
Minor cosmetical changes + agree with @FrozenPandaz's comments
